stepflow-rs/crates/stepflow-main/src/validate.rs

```diff
@@ -0,0 +1,288 @@
+#![allow(clippy::print_stdout)]
+// Licensed to the Apache Software Foundation (ASF) under one or more contributor license agreements.
+// See the NOTICE file distributed with this work for additional information regarding copyright
+// ownership.  The ASF licenses this file to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance with the License.  You may obtain a
+// copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied.  See the License for the specific language governing permissions and limitations under
+// the License.
+
+use crate::args::{load, ConfigArgs};
+use crate::{MainError, Result, stepflow_config::StepflowConfig};
+use error_stack::ResultExt as _;
+use std::path::Path;
+use stepflow_analysis::{validate_workflow, DiagnosticLevel};
+use stepflow_core::workflow::Flow;
+
+/// Validate workflow files and configuration
+///
+/// Returns the number of validation failures (errors + fatal diagnostics).
+pub async fn validate(
+    flow_path: Option<&Path>,
+    config_path: Option<&Path>,
+    config_only: bool,
+    flow_only: bool,
+) -> Result<usize> {
+    let mut failures = 0;
+
+    // Validate arguments
+    if config_only && flow_only {
+        return Err(MainError::InvalidArgument(
+            "Cannot specify both --config-only and --flow-only".to_string(),
+        )
+        .into());
+    }
+
+    if flow_only && flow_path.is_none() {
+        return Err(MainError::InvalidArgument(
+            "Must specify --flow when using --flow-only".to_string(),
+        )
+        .into());
+    }
+
+    // Validate configuration unless flow-only is specified
+    if !flow_only {
```

Comment:
Can we validate a flow without a valid config? Eg., don't we need the config to know what paths are valid, etc.?

Reply:
Oh! Good point, i was thinking just syntactic validation, but like routes need to be established, named, etc.

Reply:
I'll take out the optional arguments and just expect a stepflow-config.yml with each flow.